pkg/json2table/json2table_unit_test.go

```diff
@@ -0,0 +1,128 @@
+	InnerStructSingle *innerStruct   `json:"innerStruct,omitempty"`
+}
+
+var (
+	testStr     = "test-string"
+	testInt32   = int32(123)
+	testInt64   = int64(123543)
+	testFloat64 = float64(12.5)
+
+	innerStructJsonPaths = map[string]string{
+		"int32":  "int32",
+		"string": "string",
+	}
+
+	outerStructJsonPaths = map[string]string{
+		"string":       "string",
+		"int64":        "int64",
+		"float64":      "float64",
+		"strings":      "strings",
+		"innerStrings": "innerStructs.*.string",
+		"innerInts32":  "innerStructs.*.int32",
+	}
+
+	testInnerStruct = innerStruct{
+		Str:     &testStr,
+		Integer: &testInt32,
+	}
+
+	testOuterStruct = outerStruct{
+		Str:               &testStr,
+		Integer:           &testInt64,
+		FloatingPoint:     &testFloat64,
+		Strs:              &[]string{testStr, testStr, testStr},
+		InnerStructs:      &[]innerStruct{testInnerStruct, testInnerStruct},
+		InnerStructSingle: &testInnerStruct,
+	}
+
+	expectedResultBasicStruct = []map[string]interface{}{
+		{
+			"int32":  testInt32,
+			"string": testStr,
+		},
+	}
+
+	expectedResultComplexStruct = []map[string]interface{}{
+		{
+			"innerInts32":  []int32{testInt32, testInt32},
+			"innerStrings": []string{testStr, testStr},
+			"strings":      []string{testStr, testStr, testStr},
+			"int64":        testInt64,
+			"string":       testStr,
+			"float64":      testFloat64,
+		},
+	}
+
+	expectedResultInnerBasicStructs = []map[string]interface{}{
+		{
+			"int32":  testInt32,
+			"string": testStr,
+		},
+		{
+			"int32":  testInt32,
+			"string": testStr,
+		},
+	}
+)
+
+func TestConvertJSONToText(t *testing.T) {
+	t.Run("Convert JSON to TEXT with basic struct", testConvertJSONToTextWithBasicStruct)
+	t.Run("Convert JSON to TEXT with complex struct", testConvertJSONToTextWithComplexStruct)
+	t.Run("Convert JSON to TEXT with inner basic structs", testConvertJSONToTextWithInnerBasicStructs)
+}
+
+func testConvertJSONToTextWithBasicStruct(t *testing.T) {
+	res, err := json2table.ConvertJSONToText("", innerStructJsonPaths, testInnerStruct)
+	assert.NoError(t, err)
+	assert.Equal(t, true, compareMapSlices(res, expectedResultBasicStruct))
+}
+
+func testConvertJSONToTextWithComplexStruct(t *testing.T) {
+	res, err := json2table.ConvertJSONToText("", outerStructJsonPaths, testOuterStruct)
+	assert.NoError(t, err)
+	assert.Equal(t, true, compareMapSlices(res, expectedResultComplexStruct))
+}
+
+func testConvertJSONToTextWithInnerBasicStructs(t *testing.T) {
+	res, err := json2table.ConvertJSONToText("innerStructs", innerStructJsonPaths, testOuterStruct)
+	assert.NoError(t, err)
+	assert.Equal(t, true, compareMapSlices(res, expectedResultInnerBasicStructs))
+}
+
+func compareMapSlices(mapSlice1 []map[string]interface{}, mapSlice2 []map[string]interface{}) bool {
+	m1, err := json.Marshal(mapSlice1)
+	if err != nil {
+		return false
+	}
+
+	m2, err := json.Marshal(mapSlice2)
+	if err != nil {
+		return false
+	}
+
+	return string(m1) == string(m2)
+}
```

Comment:
I'm scared of this func - why not use `assert.Equal` directly, or `reflect.DeepEqual`?

You're marshalling an unordered collection (a map) into a serialized string comparing them. which MIGHT be problematic, I'm not sure if this might introduce any problems or not

Reply:
or if you really want to compare them in a separate func, maybe compare them key by key ?